build-tools/automation/build.linux.groovy

```diff
@@ -0,0 +1,220 @@
+// This file is based on the Jenkins scripted pipeline (as opposed to the declarative pipeline) syntax
+// https://jenkins.io/doc/book/pipeline/syntax/#scripted-pipeline
+
+def XADir = "xamarin-android"
+def buildTarget = 'jenkins'
+def chRootPackages = 'xvfb xauth mono-devel autoconf automake build-essential vim-common p7zip-full cmake gettext libtool libgdk-pixbuf2.0-dev intltool pkg-config ruby scons wget xz-utils git nuget ca-certificates-mono clang g++-mingw-w64 gcc-mingw-w64 libzip-dev openjdk-8-jdk unzip lib32stdc++6 lib32z1 libtinfo-dev:i386 linux-libc-dev:i386 zlib1g-dev:i386 gcc-multilib g++-multilib referenceassemblies-pcl zip fsharp psmisc libz-mingw-w64-dev msbuild mono-csharp-shell devscripts fakeroot debhelper libsqlite3-dev sqlite3 libc++-dev cli-common-dev mono-llvm-support curl'
+def isPr = false                // Default to CI
+def isStable = false            // Stable build workflow
+def pBuilderBindMounts = null
+def utils = null
+def hasPrLabelFullMonoIntegrationBuild = false
+
+def execChRootCommand(chRootName, chRootPackages, pBuilderBindMounts, makeCommand) {
+    chroot chrootName: chRootName,
+        additionalPackages: chRootPackages,
+        bindMounts: pBuilderBindMounts,
+        command: """
+            export LC_ALL=en_US.UTF-8
+            export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
+            locale
+
+            ${makeCommand}
+            """
+}
+
+timestamps {
+    node("${env.BotLabel}") {
+        def scmVars = null
+
+        stage ("checkout") {
+            def ctAttempts = 3
+            def retryAttempt = 0
+            def waitSecondsBeforeRetry = 15
+            retry(ctAttempts) {     // Retry will always invoke the body at least once for an attempt count of 0 or 1
+                dir (XADir) {
+                    if (retryAttempt > 0) {
+                        echo "WARNING : Stage checkout failed on try #${retryAttempt}. Waiting ${waitSecondsBeforeRetry} seconds"
+                        sleep(waitSecondsBeforeRetry)
+                        echo "Retrying ..."
+                        waitSecondsBeforeRetry = waitSecondsBeforeRetry * 2
+                    }
+
+                    retryAttempt++
+                    scmVars = checkout scm
+                }
+            }
+        }
+
+        def workspace = "/mnt/jenkins/workspace/${env.JOB_BASE_NAME}"
```

Comment:
same here

Reply:
Changed to use relative path.  Commit commit https://github.com/xamarin/xamarin-android/pull/2838/commits/316854a69cb39775d136433e566b4307a3ffe9c5